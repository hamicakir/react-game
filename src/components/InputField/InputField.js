import React, { Component } from 'react';

import styles from './InputField.module.scss';

type Props = {
  type: string,
  name: string,
  placeholder: string,
  onChange: any
}

class InputField extends Component<Props> {

  render(): React.ReactElement<any> | string | number | {} | React.ReactNodeArray | React.ReactPortal | boolean | null | undefined {
    const { type, name, placeholder, onChange } = this.props;
    return (
      <input type={type} name={name} className={styles.input} onChange={onChange} placeholder={placeholder} />
    );
  }
}

export default InputField;