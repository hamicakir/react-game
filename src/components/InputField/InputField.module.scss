.input {
  width: 100%;
  outline: 0;
  background-color: transparent;
  color: #A276AE;
  border: 0;
  border-bottom: 1px solid #363151;
  font-size: 24px;
  padding: 20px 10px;
  text-align: center;
  box-sizing: border-box;
}