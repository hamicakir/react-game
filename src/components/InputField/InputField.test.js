import React from 'react';
import renderer from 'react-test-renderer';

import InputField from './InputField';

describe('InputField Component', function () {
  it('should render correctly', () => {
    const tree = renderer.create(<InputField />).toJSON();
    expect(tree).toMatchSnapshot();
  })
});