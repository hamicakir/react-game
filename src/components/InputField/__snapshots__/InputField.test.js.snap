// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component should render correctly 1`] = `<input />`;
